ckan/controllers/package.py

```diff
@@ -1277,18 +1191,24 @@ def activity(self, id):
         data_dict = {'id': id}
         try:
             c.pkg_dict = get_action('package_show')(context, data_dict)
-            c.pkg = context['package']
-            c.package_activity_stream = get_action(
-                'package_activity_list_html')(
-                context, {'id': c.pkg_dict['id']})
             dataset_type = c.pkg_dict['type'] or 'dataset'
         except NotFound:
             abort(404, _('Dataset not found'))
         except NotAuthorized:
             abort(403, _('Unauthorized to read dataset %s') % id)
 
-        return render('package/activity.html',
-                      {'dataset_type': dataset_type})
+        return render(
+            'package/activity.html',
+            extra_vars={
+                'dataset_type': dataset_type,
```

Comment:
I see `dataset_type` being passed as extra var to templates in several places but I can't find where it is used on the actual templates. Is it necessary?

Reply:
ckanext-scheming and any site that uses multiple IDatasetForm plugins use this value, but it's not used by the core templates because they only support a single type

Reply:
But do they use it on the actual `package/activity.html` template? Why do you need to pass it to the activity templates in particular? Or is it just a convention in case someone wants to use it on their own custom `package/activity.html`?

Reply:
I include it because that's what it was doing previously, and didn't want to remove it and potentially break templates in ways I did not realize. If it can be removed that's fine with me :)

Since the entire dict is now in an activity you can switch based off the type field in the package instead.

Reply:
I agree with leaving it for customized activity pages.